src/app/api/contents/[slug]/likes/route.ts

```diff
@@ -0,0 +1,56 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { getUserHash } from 'src/app/api/shared/utils';
+
+import {
+  ContentInteraction,
+  createContentController,
+} from '~api/contents/shared/controllers';
+import { isSlugMatchingContent } from '~api/contents/shared/utils';
+import { getModelClient } from '~api/shared/model-client';
+import { ResponseBodyError } from '~api/shared/types';
+
+const client = getModelClient();
+
+interface PostBody {
+  value?: number;
+}
+
+export async function POST(
+  req: NextRequest,
+  { params }: { params: { slug: string } }
+): Promise<NextResponse<ContentInteraction> | NextResponse<ResponseBodyError>> {
+  const { slug } = params;
+
+  const body = (await req.json()) as PostBody;
+  const { value } = body;
+
+  if (!value) {
+    return NextResponse.json(
+      { error: 'Missing `value` in the request body.' },
+      { status: 400 }
+    );
+  }
+
+  const isValidSlug = await isSlugMatchingContent(slug);
+  if (!isValidSlug) {
+    return NextResponse.json(
+      { error: `Slug "${slug}" not found` },
+      { status: 400 }
+    );
+  }
+
+  const userIp = req.ip ?? '127.0.0.1';
+  if (!userIp) {
+    return NextResponse.json(
+      { error: `Missing 'ip' in the request` },
+      { status: 400 }
+    );
+  }
```

Comment:
TODO: shared logic: move into a middleware; userHash should already be populated into the request